package.json

```diff
@@ -6,51 +6,52 @@
   "license": "Apache-2.0",
   "homepage": "https://github.com/softonic/axios-retry",
   "files": [
-    "es",
-    "lib",
-    "index.js",
-    "index.d.ts"
+    "dist"
   ],
   "scripts": {
-    "lint": "eslint es/**/*.mjs spec/**/*.spec.mjs",
+    "lint": "eslint .",
     "pretest": "npm run lint",
-    "test": "NODE_OPTIONS=--es-module-specifier-resolution=node jasmine",
+    "test": "ts-node --project tsconfig-cjs.json node_modules/jasmine/bin/jasmine",
     "prebuild": "npm run test",
-    "build": "rm -rf lib && babel es -d lib/esm --source-maps && babel es -d lib/cjs --config-file ./babel.config.cjs.json --source-maps && ./fixup",
+    "build": "rm -fr dist/* && npm run build:esm && npm run build:cjs",
+    "build:esm": "tsc -p tsconfig.json && echo '{\"type\":\"module\"}' > dist/esm/package.json",
+    "build:cjs": "tsc -p tsconfig-cjs.json && echo '{\"type\":\"commonjs\"}' > dist/cjs/package.json",
```

Comment:
I find this creation of `package.json` in subdirectories weird and hacky. I have never seen it being done elsewhere.

Since the `exports` object defines whether something is esm or cjs, I don't get why it would be necessary to also add this `package.json`.

I saw this approach mentioned in the guide linked in this PR but it's not really explained much there and I haven't seen it used anywhere else.

Do you know of any specific case that would fail without it?

Reply:
While certainly a bit hackneyed, this is one effective method. This method is https://nodejs.org/api/packages.html#dual-package-hazard. The following is a summary of how the Dual Package can be achieved and is informative.

https://github.com/microsoft/TypeScript/issues/49462#issuecomment-1633279027

And this method is also taken by other libraries such as [stripe-node](https://github.com/stripe/stripe-node).

> Do you know of any specific case that would fail without it?

If we do not do this, even if you separate them into directories esm, cjs, the module system is still determined by the module settings in package.json.
Therefore, you will get the following compile error.

```
This expression is not callable.
  Type 'typeof import("/home/study/workspace/ts-node-oidc/node_modules/axios-retry/dist/esm/index")' has no call signatures.ts(2349)
```
![image](https://github.com/softonic/axios-retry/assets/79501292/9eafe9c6-d0f2-4939-a2f0-fc04b6b9ea5a)

This error occurs when deleting package.json from esm in `node_module`.

![image](https://github.com/softonic/axios-retry/assets/79501292/b597fa8d-e5a3-43eb-b881-9f7966bc9ac9)

Reply:
OK. I trust what you are saying.
But just gonna note that the official Node documentation doesn't mention those extra `package.json`'s so it's weird to me that that would be needed.

I wanted to also verify it on my own but I could not reproduce it on my side with this setup and extra `package.json`'s removed:

![Screenshot 2023-11-22 at 09 14 13](https://github.com/softonic/axios-retry/assets/153197/41db4020-2891-4bb0-b154-52d5755b04fd)

But I can see that Typescript insists on importing from `/cjs/` so that's probably why but I'm not sure why it's doing that...